### PR TITLE
dotfiles: miscellaneous improvements

### PR DESCRIPTION
**Pull request by akhilnarang**

### CODE REVIEW

File: common [hunk 1]
```diff
@@ -23,17 +23,23 @@ function clip {
...
 
 function setname {
     echo -ne "\033]0;${1}\007"
 }
 
 function get_latest_release() {
-  curl --silent "https://api.github.com/repos/$1/releases/latest" | # Get latest release from GitHub api
-    grep '"tag_name":' |                                            # Get tag line
-    sed -E 's/.*"([^"]+)".*/\1/'                                    # Pluck JSON value
+  curl --silent "https://api.github.com/repos/$1/releases/latest" | # Get latest release from GitHub API
```
Comment (msfjarvis) [inline common]: Change `$1` to `${1:?}` to enforce its presence.

File: common [hunk 2]
```diff
@@ -23,17 +23,23 @@ function clip {
...
+  jq -r .tag_name                                                   # Get tag line
+}
+
+function get_latest_release_assets() {
+  local id=$(curl --silent "https://api.github.com/repos/$1/releases/latest" | # Get the latest release from GitHub API
```
Comment (msfjarvis) [inline common]: Ditto

File: common [hunk 3]
```diff
@@ -4,7 +4,7 @@
 # SPDX-License-Identifier: GPL-3.0-only
 
 # Import all the bot tokens and IDs
-# shellcheck disable=SC1090
+# shellcheck disable=SC1090,SC1117,SC2155
```
Comment (msfjarvis) [inline common]: Why these?

**Review by msfjarvis — CHANGES_REQUESTED** [review]

File: common (hunk 1 above)
Comment (akhilnarang) [reply common]: Okay

**Review by msfjarvis — DISMISSED** [review]

**Comment by msfjarvis** [comment]
```
msfjarvis@ryzenbox ~/git-repos/lazy-bash (master) $ get_latest_release MSF-Jarvis/oneplus3
ShockableSatsuma
msfjarvis@ryzenbox ~/git-repos/lazy-bash (master) $ get_latest_release_assets MSF-Jarvis/oneplus3
jq: error (at <stdin>:4): Cannot iterate over null (null)
```

Doesn't work for me

**Pull request merged**